src/Codeception/Test/Loader.php

```diff
@@ -0,0 +1,134 @@
+<?php
+namespace Codeception\Test;
+
+use Codeception\Test\Loader\Cept as CeptLoader;
+use Codeception\Test\Loader\Cest as CestLoader;
+use Codeception\Test\Loader\TestCase as TestCaseLoader;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * Loads all Codeception supported test formats from a directory.
+ *
+ * ``` php
+ * <?php
+ * $testLoader = new \Codeception\TestLoader('tests/unit');
+ * $testLoader->loadTests();
+ * $tests = $testLoader->getTests();
+ * ?>
+ * ```
+ * You can load specific file
+ *
+ * ``` php
+ * <?php
+ * $testLoader = new \Codeception\TestLoader('tests/unit');
+ * $testLoader->loadTest('UserTest.php');
+ * $testLoader->loadTest('PostTest.php');
+ * $tests = $testLoader->getTests();
+ * ?>
+ * ```
+ * or a subdirectory
+ *
+ * ``` php
+ * <?php
+ * $testLoader = new \Codeception\TestLoader('tests/unit');
+ * $testLoader->loadTest('models'); // all tests from tests/unit/models
+ * $tests = $testLoader->getTests();
+ * ?>
+ * ```
+ *
+ */
+class Loader
+{
+    protected $formats = [];
+    protected $tests = [];
+    protected $path;
+
+    public function __construct(array $suiteSettings)
+    {
+        $this->path = $suiteSettings['path'];
+        $this->formats = [
+            new CeptLoader(),
+            new CestLoader(),
+            new TestCaseLoader(),
+        ];
+    }
+
+    public function getTests()
+    {
+        return $this->tests;
+    }
+
+    protected function relativeName($file)
+    {
+        return str_replace([$this->path, '\\'], ['', '/'], $file);
+    }
+
+    protected function findPath($path)
+    {
+        if (!file_exists($path)
+            && substr(strtolower($path), -strlen('.php')) !== '.php'
```

Comment:
Why do you use strtolower here if all other filename matching is case-sensitive?
I would use -4  instead of -strlen().